hibp.js

```diff
@@ -76,9 +78,32 @@ const HIBP = {
     } catch (error) {
       console.error(error);
     }
-    return foundBreaches.filter(breach => breach.IsVerified && !breach.IsRetired && !breach.IsSensitive && !breach.IsSpamList);
+    return this.filterOutUnsafeBreaches(foundBreaches);
   },
 
+
+  filterOutUnsafeBreaches(breaches) {
+    return breaches.filter(
+      breach => breach.IsVerified && 
+      !breach.IsRetired && 
+      !breach.IsSensitive &&
+      !breach.IsSpamList
+    );
+  },
+
+
+  getLatestBreachDateTime(breaches) {
```

Comment:
We seem to have quite a few functions that loop through the entire list of breaches. How about doing a single loop in HIBP.js when we fetch breaches, and storing all the stuff we need there so that we don't have to keep looping later?

Reply:
I started to code this, but it turned into a significant refactor so I'd like to do this as a follow-up issue. I'll push a branch up.

Reply:
Ref: #290